megamek/src/megamek/client/ui/swing/boardview/FiringArcSpriteHandler.java

```diff
@@ -0,0 +1,415 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.swing.boardview;
+
+import megamek.client.ui.swing.*;
+import megamek.common.*;
+import megamek.common.equipment.WeaponMounted;
+import megamek.common.options.OptionsConstants;
+import megamek.common.preference.IPreferenceChangeListener;
+import megamek.common.preference.PreferenceChangeEvent;
+import megamek.common.weapons.infantry.InfantryWeapon;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This BoardViewSpriteHandler handles the sprites for the firing arcs (field of fire) that can be shown for
```

Comment:
*Thank you* for putting "firing arc" in the comments; "field of fire" is not intuitive to me and I'm constantly struggling to find this kind of code because of it.

Reply:
I was having trouble finding the GUIP setting myself :D And I added that whole feature originally (was merged at the time by arlith as I had no clue about PRs. Ten years ago)

Reply:
Pretty sure it actually predated GitHub when we where on Sourceforge and you had to do patches via Subversion.  You're patch generated a lot of buzz in the dev team at that time.